src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1081,62 +1081,52 @@ public StringBuilder Append(char[] value)
             return this;
         }
 
-        // Append joined values with a separator between each value.
-        public unsafe StringBuilder AppendJoin<T>(char separator, params T[] values)
-        {
-            // Defer argument validation to the internal function
-            return AppendJoinCore(&separator, 1, values);
-        }
+        
+        #region AppendJoin
 
-        public unsafe StringBuilder AppendJoin<T>(string separator, params T[] values)
+        public unsafe StringBuilder AppendJoin(string separator, params object[] values)
         {
             separator = separator ?? string.Empty;
             fixed (char* pSeparator = separator)
             {
-                // Defer argument validation to the internal function
                 return AppendJoinCore(pSeparator, separator.Length, values);
             }
         }
 
-        public unsafe StringBuilder AppendJoin<T>(char separator, IEnumerable<T> values)
+        public unsafe StringBuilder AppendJoin<T>(string separator, IEnumerable<T> values)
         {
-            // Defer argument validation to the internal function
-            return AppendJoinCore(&separator, 1, values);
+            separator = separator ?? string.Empty;
+            fixed (char* pSeparator = separator)
+            {
+                return AppendJoinCore(pSeparator, separator.Length, values);
+            }
         }
 
-        public unsafe StringBuilder AppendJoin<T>(string separator, IEnumerable<T> values)
+        public unsafe StringBuilder AppendJoin(string separator, params string[] values)
```

Comment:
We care, but I think `values` makes much more sense. I think we should follow `String.Join<T>` which used `values` as well:

https://apisof.net/catalog/System.String.Join%3CT%3E(String,IEnumerable%3CT%3E)